chromium_src/net/dns/dns_transaction.cc

```diff
@@ -0,0 +1,10 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "net/base/load_flags.h"
+
+#define LOAD_BYPASS_PROXY LOAD_DISABLE_CACHE
```

Comment:
I really don't think we should do this for all requests. I checked git history around this code (https://chromium-review.googlesource.com/c/chromium/src/+/1843334/10/net/dns/dns_transaction.cc), and the authors say LOAD_BYPASS_PROXY is here to prevent deadlocks.

So we should either try upstreaming this or limit to tor-related requests

Reply:
That is for `SetDisableSecureDns`
It's been there since the first DoH commit 
https://chromium-review.googlesource.com/c/chromium/src/+/710554/40/net/dns/dns_transaction.cc#431

Reply:
yea, I actually can't really imagine why we'd ever want to bypass the proxy for DoH requests. That seems like it would actually prevent requests from working at all if the proxy was required for access and non-proxy requests are blocked by firewall

Reply:
There is a discussion with upstream net dev
https://groups.google.com/u/1/a/chromium.org/g/net-dev/c/GI4E9ERwPgw

Reply:
proxy hostname resolution will go through DoH if DoH is enabled
<img width="719" alt="Screen Shot 2021-02-03 at 16 19 24" src="https://user-images.githubusercontent.com/11330831/106826489-df46e100-663b-11eb-86eb-637d4ed87a6a.png">